### PR TITLE
[compiler-rt] adding preadv2/pwritev2 interceptions.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc [hunk 1]
```diff
@@ -10259,6 +10259,38 @@ INTERCEPTOR(int, cpuset_getaffinity, int level, int which, __int64_t id, SIZE_T
 #define INIT_CPUSET_GETAFFINITY
 #endif
 
+#if SANITIZER_INTERCEPT_PREADV2
+INTERCEPTOR(SSIZE_T, preadv2, int fd, __sanitizer_iovec *iov, int iovcnt,
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_common_interceptors.inc:10263]: can you please add simple tests in sanitizer_common?

**Review by vitalybuka — APPROVED** [review]

**Comment by mysterymath** [comment]
This broke the Fuchsia toolchain build:
https://luci-milo.appspot.com/ui/p/fuchsia/builders/toolchain.ci/clang-linux-x64/b8742307122405229761/overview

Reverted, but let me know if the fix doesn't end up being obvious.

**Comment by devnexen** [comment]
How should I disable the test for fuchsia ? it required glibc though, weird..

**Comment by mysterymath** [comment]
We're not running the test on Fuchsia, it's run on Fuchsia as part of the host build. Our glibc version is old enough that it doesn't include preadv2 or pwritev2. There needs to be a CMake compilation check that enables or disables the test.

**Pull request merged**